h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -344,15 +352,15 @@ public GuessSetupTsk(ParseSetup userSetup) {
 
         // Compute the max line length (to help estimate the number of bytes to read per Parse map)
         _maxLineLength = maxLineLength(bits);
-        if (_maxLineLength==-1) throw new H2OIllegalArgumentException("The first 4MB of the data don't contain any line breaks. Cannot parse.");
 
-        // only preview 1 DFLT_CHUNK_SIZE for ByteVecs, UploadFileVecs, compressed, and small files
-/*        if (ice instanceof ByteVec
-                || ((Frame)ice).vecs()[0] instanceof UploadFileVec
-                || bv.length() <= FileVec.DFLT_CHUNK_SIZE
-                || decompRatio > 1.0) { */
         try {
           _gblSetup = guessSetup(bv, bits, _userSetup);
+          _gblSetup.bytesStats = new BytesStats(bits);
+          {
+            if (_maxLineLength != _gblSetup.bytesStats.maxWidth) {
```

Comment:
Look at the next line, it's commented out. There's something fishy in data handling. This is the next thing to investigate. Let's keep it for a while. Once I add non-ascii data and reopen the case with wrong line length evaluation, it will do the right stuff.
Commenting it out for now.